Sources/FoundationEssentials/String/String+Internals.swift

```diff
@@ -19,6 +19,65 @@ internal import Foundation_Private.NSString
 import Darwin
 #endif
 
+#if os(Windows)
+import WinSDK
+
+extension String {
+    package func withNTPathRepresentation<Result>(_ body: (UnsafePointer<WCHAR>) throws -> Result) throws -> Result {
+        guard !isEmpty else {
+            throw CocoaError.errorWithFilePath(.fileReadInvalidFileName, "")
+        }
+
+        // 1. Normalize the path first.
+
+        var path = self
+
+        // Strip the leading `/` on a RFC8089 path (`/[drive-letter]:/...` ).  A
+        // leading slash indicates a rooted path on the drive for the current
+        // working directory.
+        var iter = path.makeIterator()
+        if iter.next() == "/", iter.next()?.isLetter ?? false, iter.next() == ":" {
+            path.removeFirst()
+        }
+
+        // Win32 APIs can support `/` for the arc separator. However,
+        // symlinks created with `/` do not resolve properly, so normalize
+        // the path.
+        path.replace("/", with: "\\")
+
+        // Drop trailing slashes unless it follows a drive specification.  The
+        // trailing arc separator after a drive specifier indicates the root as
+        // opposed to a drive relative path.
+        let first = path.startIndex
+        let second = path.index(after: path.startIndex)
```

Comment:
Sorry to nitpick one more time but I think this one is important: can we move these index calculations into the `while` loop? Any mutation performed by the `path.removeLast()` call will invalidate any pre-existing indices, so we need to re-calculate `first` and `second` after mutating the string each time since the indices may be invalid

Reply:
Done